packages/mcp/src/configuration.ts

```diff
@@ -21,10 +29,32 @@ export class MCPConfiguration extends MastraBase {
     super({ name: 'MCPConfiguration' });
     this.defaultTimeout = args.timeout ?? DEFAULT_REQUEST_TIMEOUT_MSEC;
     this.serverConfigs = args.servers;
-    this.id = args.id ?? this.makeId();
 
-    // to prevent memory leaks return the same MCP server instance when configured the same way multiple times
-    const existingInstance = mastraMCPConfigurationInstances.get(this.id);
+    // If an ID is provided, use it directly
+    if (args.id) {
+      this.id = args.id;
+      const cached = (this.constructor as typeof MCPConfiguration).serverConfigCache.get(this.id);
```

Comment:
What's the reason for moving this to a static property? And if we need it why not this:
```suggestion
      const cached = MCPConfiguration.serverConfigCache.get(this.id);
```

Reply:
I was testing that possibility, should have not included it on the PR on the first place. My bad 😅